granary/flickr.py

```diff
@@ -0,0 +1,371 @@
+      params['extras'] = self.API_EXTRAS
+      params['per_page'] = 50
+
+    assert method
+    logging.debug('calling %s with %s', method, params)
+    photos_resp = self.call_api_method(method, params)
+
+    result = {'items': []}
+    if solo:
+      photos = [photos_resp.get('photo', {})]
+    else:
+      photos = photos_resp.get('photos', {}).get('photo', [])
+
+    for photo in photos:
+      activity = self.photo_to_activity(photo)
+
+      # TODO consider using 'flickr.activity.userPhotos' when group_id=@self,
+      # gives all recent comments and faves, instead of hitting the API for
+      # each photo
+      if fetch_replies:
+        replies = []
+        comments_resp = self.call_api_method('flickr.photos.comments.getList', {
+          'photo_id': photo.get('id'),
+        })
+        for comment in comments_resp.get('comments', {}).get('comment', []):
+          replies.append(self.comment_to_object(comment, photo.get('id')))
+          activity['object']['replies'] = {
+            'items': replies,
+            'totalItems': len(replies),
+          }
+
+      if fetch_likes:
+        faves_resp = self.call_api_method('flickr.photos.getFavorites', {
+          'photo_id': photo.get('id'),
+        })
+        for person in faves_resp.get('photo', {}).get('person', []):
+          activity['object'].setdefault('tags', []).append({
+            'author': {
+              'objectType': 'person',
+              'displayName': person.get('realname'),
+              'username': person.get('username'),
+              'id': self.tag_uri(person.get('nsid')),
+              'image': {
+                'url': self.get_user_image(person.get('iconfarm'),
+                                           person.get('iconserver'),
+                                           person.get('nsid')),
+              },
+            },
+            'created': self.reformat_unix_time(activity.get('favedate')),
+            'url': '{}#liked-by-{}'.format(
+              activity.get('url'), person.get('nsid')),
+            'object': {'url': activity.get('url')},
+            'id': self.tag_uri('{}_liked_by_{}'.format(
+              photo.get('id'), person.get('nsid'))),
+            'objectType': 'activity',
+            'verb': 'like',
+          })
+
+      result['items'].append(activity)
+
+    return util.trim_nulls(result)
+
+  def get_actor(self, user_id=None):
+    """Get an ActivityStreams object of type 'person' given a Flickr user's nsid.
+    If no user_id is provided, this method will make another API requeset to
+    find out the currently logged in user's id.
+
+    Args:
+      user_id: string, optional
+
+    Returns:
+      dict, an ActivityStreams object
+    """
+    if not user_id:
+      user_id = self.user_id
+
+    if not user_id:
+      resp = self.call_api_method('flickr.people.getLimits')
+      user_id = resp.get('person', {}).get('nsid')
+
+    logging.debug('calling flickr.people.getInfo with user_id %s', user_id)
```

Comment:
yay logging! maybe consider moving this to `call_api_method()` though, maybe even the one inside oauth-dropins, so it's reused